release-notes.txt

```diff
@@ -1,3 +1,780 @@
++ at86rf2xx: add support for high data rates
++ dfplayer: New driver for the DFPlayer Mini MP3 player
+* ethos: use luid_get_eui48()
++ edbg_eui: add driver to get MAC address from Atmel EDBG
++ mhz19: initial support
++ mrf24j40: add Turbo Mode
++ mtd_spi_eeprom: add MTD wrapper for AT25XXX EEPROMs
++ pca9633: add new driver
++ periph/timer: add timer_set_periodic()
++ periph/i2c: add periph_i2c_reconfigure feature & implementation for sam0
++ periph/rtt: introduce RTT_MIN_OFFSET
++ periph/spi: add periph_spi_reconfigure feature & implementation for sam0
+* pulse_counter: Use C11 atomics & bugfix
++ sdp3x : Add driver for sdp3x sensor
+
+Build System / Tooling
+----------------------
++ make:
+  + add code generator targets
+  + add JLink support for stm32
+  * docker.inc.mk: add and use generic docker_run_make
+  + allow multiple external board dirs
+  + info-global.inc.mk: add info-boards-features-blacklisted
+  * resolve dependencies before Makefile.include
++ riotctrl_shell:
+  + initial import of shell interaction spawn
+  + provide ipv6_nib interactions and parsers
+  + provide netif interactions and parsers
++ tools:
+  + add script to run radvd on a tun interface
+  + provide tool to parse `pktbuf` output
+  * kconfiglib:
+    * add riot_kconfig to override default behaviours
+    * kconfiglib: bump to v14.1.0
+
+Testing
+-------
++ fuzzing framework:
+  + add AFL-based fuzzing setup for network modules
+  + add fuzzing application for gcoap
++ ci: setup github actions
++ sys/test_utils/dummy_thread: initial commit
++ tests/irq_disable_restore: Add test for irq_disable() / irq_restore()
++ tests/periph_spi: add benchmark for acquire / release
++ tests/periph_spi_dma: Add test for SPI with DMA
++ uncrustify: add auto uncrustify with blacklist
+
+Kconfig migration
+-----------------
++ armv7_m/cortexm: declare CPU_ARCH and CPU_CORE
++ boards: Model features in Kconfig
+  + arduino-leonardo, arduino-mega2560, atmega128x-based, atmega256rfr2-based,
+    atmega328p-based, cc2538-based, cc26xx_cc13xx-based, efm32-based, esp-based,
+    fe310-based, kinetis-based, msp430-based, samr21-xpro, slwstk6220a
+* kconfig: Declare common feature symbols
++ drivers: Expose Configurations to Kconfig
+  + ads101x, at, at86rf215, bmx055, cc110x, dose.h, fxos8700, hdc1000, isl29020,
+    l3g4200d, lpsxxx, mag3110, mma8x5x, motor_driver, mtd_sdcard, opt3001,
+    pn532, rn2xx3, slipdev, sps30, tcs37727, tmp00x
++ networking: Expose configurations to Kconfig
+  + asymcute, coap, csma
+  + gnrc:
+    + gomach, lwmac, mac, pktdump, pktbuf, rpl, tcp
++ usbus/cdc/ecm: Expose configurations to Kconfig
+
+API Changes
+===========
+- cpu/sam0_common: drop prescaler from timer config
+- cpu/sam0_common: PWM config now requires GCLK source
+- drivers/mtd: Change API to return 0 on success
+- drivers/mtd_mapper: count offset in sectors
+- net/sock_dtls: provide getter function for UDP sock
+- net/sock_dtls: add timeout to sock_dtls_send and add sock_dtls_session_init
+
```

Comment:
Does this large diff belong here? It's way more detailed than all the rest, and looks a bit like a copy-pasting slip.

Reply:
The rest are either simple changes or additions. This change is a larger API overhaul and thus justifies a more detailed listing of the changes.

Reply:
(see also larger API changes in other releases below)